lib/hanami/commands/generate/model.rb

```diff
@@ -103,11 +103,24 @@ def repository_path
         # @since 0.9.1
         # @api private
         def migration_path
+          existing_migration_path || new_migration_path
+        end
+
+        # @since 1.1.0
+        # @api private
+        def new_migration_path
           timestamp = Time.now.utc.strftime(Migration::TIMESTAMP_FORMAT)
-          filename = Migration::FILENAME_PATTERN % { timestamp: timestamp, name: "create_#{table_name}"}
+          filename = Migration::FILENAME_PATTERN % { timestamp: timestamp, name: migration_name_underscored }
+
           Hanami::Model.configuration.migrations.join(filename)
         end
 
+        # @since 1.1.0
+        # @api private
+        def existing_migration_path
+          Utils::FileList["#{Hanami::Model.configuration.migrations}/[0-9]*_#{migration_name_underscored}.rb"].first
```

Comment:
@kaikuchn 

Thanks for your thoughts. This code was here before my changes, but you're right - this is little mismatch. 

We can do something like 
```
Hanami::Utils::FileList["db/migrations/*.rb"].find { |filename| %r{/\d{14,}_create_campaigns.rb\z} =~ filename }
```

This is looks more complex, but works more accurately.